src/configuration-client/src/main/java/ee/ria/xroad/common/conf/globalconf/FederationConfigurationSourceFilterImpl.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The MIT License
+ * Copyright (c) 2015 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package ee.ria.xroad.common.conf.globalconf;
+
+import ee.ria.xroad.common.SystemProperties;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+
+import static ee.ria.xroad.common.SystemProperties.AllowedFederationMode.*;
+
+/** Implementation of the {@link FederationConfigurationSourceFilter}.
+ * Looks at {@link SystemProperties#getConfigurationClientAllowedFederations()} (a comma-separated list of allowed
+ * instances).
+ * <ol>
+ * <li>If it contains {@link ee.ria.xroad.common.SystemProperties.AllowedFederationMode#NONE}, it will return
+ * {@code false} for all {@link #shouldDownloadConfigurationFor(String)} queries.</li>
+ * <li>If previous is untrue but the string contains
+ * {@link ee.ria.xroad.common.SystemProperties.AllowedFederationMode#ALL}, it will return {@code true} for all
+ * queries.</li>
+ * <li>If previous are untrue, it will return {@code true} if the list contains the (case-insensitive) X-Road instance
+ * name</li>
+ * </ol>
+ *
+ */
+@Slf4j
+public class FederationConfigurationSourceFilterImpl implements FederationConfigurationSourceFilter {
+
+    private static final String COMMA_SEPARATOR = "\\s*,\\s*";
+
+    private final String ownInstance;
+
+    private SystemProperties.AllowedFederationMode allowedFederationMode = null;
+    private Set<String> allowedFederationPartners = null;
+
+    FederationConfigurationSourceFilterImpl(String ownInstance) {
+        this.ownInstance = ownInstance;
+        String filterString = SystemProperties.getConfigurationClientAllowedFederations();
+        log.info("The federation filter system property value is: '{}'", filterString);
+        allowedFederationPartners = parseAllowedInstances(Arrays.asList(filterString.split(COMMA_SEPARATOR)));
```

Comment:
parseAllowedInstances sets the allowedFederationMode field as a side effect. Consider removing the return value and set both fields in the method or removing the side effect and making both fields final.